src/libcore/iter/iterator.rs

```diff
@@ -30,11 +30,56 @@ fn _assert_is_object_safe(_: &dyn Iterator<Item=()>) {}
 /// [impl]: index.html#implementing-iterator
 #[stable(feature = "rust1", since = "1.0.0")]
 #[rustc_on_unimplemented(
+    on(
+        _Self="[std::ops::Range<Idx>; 1]",
+        label="if you meant to iterate between two values, remove the square brackets",
+        note="`[start..end]` is an array of one `Range`; you might have meant to have a `Range` \
+              without the brackets: `start..end`"
+    ),
+    on(
+        _Self="[std::ops::RangeFrom<Idx>; 1]",
+        label="if you meant to iterate from a value onwards, remove the square brackets",
+        note="`[start..]` is an array of one `RangeFrom`; you might have meant to have a \
+              `RangeFrom` without the brackets: `start..`"
+    ),
+    on(
+        _Self="[std::ops::RangeTo<Idx>; 1]",
+        label="if you meant to iterate until a value, remove the square brackets",
```

Comment:
You can't iterate over a `RangeTo`, so I'm not sure this one is necessary.

Reply:
You're absolutely right, I just never had thought about it enough to look it up. I will still add a note to it in order to let users know that.